netsim/modules/vlan.py

```diff
@@ -1280,6 +1280,29 @@ def link_pre_transform(self, link: Box, topology: Box) -> None:
   def module_pre_link_transform(self, topology: Box) -> None:
     create_loopback_vlan_links(topology)
 
+  # We need to copy a few attributes from node VLAN data into access link interface data, for example
+  # the node IPv4/IPv6 address (to support static addressing) and "gateway" attribute to enable
+  # selective VLAN gateway module activation
+  #
+  # This operation needs to be done when the link data structures have already been transformed into
+  # the canonical "interfaces" list format, so we cannot do it any sooner than this point.
+  #
+  def link_pre_link_transform(self, link: Box, topology: Box) -> None:
+    if 'vlan' not in link:
+      return
+
+    for intf in link.interfaces:
+      vname = intf.get('vlan.access',None)
+      if not vname:
+        continue
+      if vname not in topology.nodes[intf.node].get('vlans',{}):
+        continue
+      vdata = topology.nodes[intf.node].vlans[vname]
+      for attr in topology.defaults.vlan.attributes.copy_vlan_to_intf:
```

Comment:
Do you anticipate that other modules might need to extend this list, and that's why it's defined in ```vlan.yml```?
If not, I would just list the attributes here:
```
for attr in ["ipv4","ipv6","gateway"]
```

In particular for the ```gateway``` attribute, it would be "better" (more elegant imho) to have the ```gateway``` module code append "gateway" to the list of attributes to be copied. As is, the ```vlan``` module is making assumptions about another module

Reply:
I love having stuff like that list defined in the defaults (as opposed to hardcoded inline or defined in the module header) to make it easier to change if we hit a snag, similar to how I could write the instructions for fixing containerlab startup parameters (blog post coming in an hour or two)